src/components/tooltip/TooltipManager.ts

```diff
@@ -28,15 +30,25 @@ export default class TooltipManager {
 
   registerElement(referenceEl: ReferenceElement, tooltip: HTMLCalciteTooltipElement): void {
     this.registeredElementCount++;
-
     this.registeredElements.set(referenceEl, tooltip);
+    const shadowRoot = this.getReferenceElShadowRootNode(referenceEl);
```

Comment:
This works assuming there's only one layer of shadow DOM, right? I believe the following would still have issues:

```html
<composite-component-a>
  <!-- shadow DOM -->
    <composite-component-b>
        <!-- shadow DOM -->
           <calcite-link id="tooltip-button">Data disclaimer</calcite-link>
           <calcite-tooltip label="Data disclaimer" reference-element="tooltip-button">
             <span>This data was collected over a 24 hour period</span>
           </calcite-tooltip>
        <!-- shadow DOM -->    
    </composite-component-b>    
  <!-- shadow DOM -->
</composite-component-a>
```

Reply:
That's a good question, I haven't tested that but I would have assumed it should work since this just has to deal with focusing and once you focus into the 2nd layer of shadowDom the event would be triggered right?

Reply:
For the example above, the `getReferenceElShadowRootNode` should return the 2nd layer shadow DOM right? Since the rootNode for that component would be the 2nd layer shadow DOM. right?

Reply:
This does work:

```html
<template shadowrootmode="open">
    <div>
      <template shadowrootmode="open">
        <calcite-button id="shadow-trigger-1">
          <calcite-icon icon="grid" />
        </calcite-button>
        <calcite-tooltip reference-element="shadow-trigger-1" placement="bottom">
          Display Grid
        </calcite-tooltip>
        <calcite-button id="shadow-trigger-2">
          <calcite-icon icon="list" />
        </calcite-button>
        <calcite-tooltip reference-element="shadow-trigger-2" placement="bottom">
          Display List
        </calcite-tooltip>
      </template>
    </div>
  </template>
```

Reply:
Tested and seems to work as expected. Getting the `getReferenceElShadowRootNode` will get the closest shadowRoot and adding focusIn/focusout events on that shadowRoot works since anything inside that shadowRoot getting focus will trigger the events.

Reply:
Added test

Reply:
Awesome, thanks for testing!